rasa/telemetry.py

```diff
@@ -0,0 +1,553 @@
+import rasa.utils.io
+
+logger = logging.getLogger(__name__)
+
+SEGMENT_ENDPOINT = "https://api.segment.io/v1/track"
+
+TELEMETRY_ENABLED_ENVIRONMENT_VARIABLE = "RASA_TELEMETRY_ENABLED"
+TELEMETRY_DEBUG_ENVIRONMENT_VARIABLE = "RASA_TELEMETRY_DEBUG"
+
+# the environment variable can be used for local development to set a test key
+# e.g. `RASA_TELEMETRY_WRITE_KEY=12354 rasa train`
+TELEMETRY_WRITE_KEY_ENVIRONMENT_VARIABLE = "RASA_TELEMETRY_WRITE_KEY"
+
+TELEMETRY_ID = "metrics_id"
+TELEMETRY_ENABLED_BY_DEFAULT = True
+
+# if one of these environment variables is set, we assume to be running in CI env
+CI_ENVIRONMENT_TELL = [
+    "bamboo.buildKey",
+    "BUILD_ID",
+    "BUILD_NUMBER",
+    "BUILDKITE",
+    "CI",
+    "CIRCLECI",
+    "CONTINUOUS_INTEGRATION",
+    "GITHUB_ACTIONS",
+    "HUDSON_URL",
+    "JENKINS_URL",
+    "TEAMCITY_VERSION",
+    "TRAVIS",
+]
+
+# If updating or creating a new event, remember to update
+# https://rasa.com/docs/rasa/telemetry
+TRAINING_STARTED_EVENT = "Training Started"
+TRAINING_COMPLETED_EVENT = "Training Completed"
+TELEMETRY_DISABLED = "Telemetry Disabled"
+
+
+def print_telemetry_reporting_info() -> None:
+    """Print telemetry information to std out."""
+    message = textwrap.dedent(
+        f"""
+      Rasa reports anonymous usage telemetry to help improve Rasa Open Source
+      for all its users.
+
+      If you'd like to opt-out, you can use `rasa telemetry disable`.
+      To learn more, check out {DOCS_URL_TELEMETRY}."""
+    ).strip()
+
+    table = SingleTable([[message]])
+    print(table.table)
+
+
+def _default_telemetry_configuration(is_enabled: bool) -> Dict[Text, Text]:
+    return {
+        CONFIG_TELEMETRY_ENABLED: is_enabled,
+        CONFIG_TELEMETRY_ID: uuid.uuid4().hex,
+        CONFIG_TELEMETRY_DATE: datetime.now(),
+    }
+
+
+def _write_default_telemetry_configuration(
+    is_enabled: bool = TELEMETRY_ENABLED_BY_DEFAULT,
+) -> None:
+    if is_enabled:
+        print_telemetry_reporting_info()
+
+    new_config = _default_telemetry_configuration(is_enabled)
+
+    rasa_utils.write_global_config_value(CONFIG_FILE_TELEMETRY_KEY, new_config)
+
+
+def _is_telemetry_enabled_in_configuration() -> bool:
+    """Read telemetry configuration from the user's Rasa config file in $HOME.
+
+    Creates a default configuration if no configuration exists.
+
+    Returns:
+        `True`, if telemetry is enabled, `False` otherwise.
+    """
+    try:
+        stored_config = rasa_utils.read_global_config_value(
+            CONFIG_FILE_TELEMETRY_KEY, unavailable_ok=False
+        )
+
+        return stored_config[CONFIG_TELEMETRY_ENABLED]
+    except ValueError as e:
+        logger.debug(f"Could not read telemetry settings from configuration file: {e}")
+
+        # seems like there is no config, we'll create on and enable telemetry
+        _write_default_telemetry_configuration()
+        return TELEMETRY_ENABLED_BY_DEFAULT
+
+
+def initialize_telemetry() -> bool:
```

Comment:
```suggestion
def _initialize_telemetry() -> bool:
```

Reply:
used outside of this module so shouldn't be private

Reply:
Sorry 🙈  for some reason pycharm didn't show this usage 🤔